src/Components/Components/perf/RouteTableFactoryBenchmarks.cs

```diff
@@ -0,0 +1,2833 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components.Performance;
+
+public class RouteTableFactoryBenchmarks
+{
+    private IServiceProvider _serviceProvider;
+    private Dictionary<Type, string[]> _templatesByHandler;
+
+    public RouteTableFactoryBenchmarks()
+    {
+        _serviceProvider = new ServiceCollection()
+            .AddLogging()
+            .BuildServiceProvider();
+
+        _templatesByHandler = new Dictionary<Type, string[]>
+        {
+            { typeof(object), _routes }
+        };
+    }
+
+    [Benchmark]
+    public object CreateRouteTable() => RouteTableFactory.Create(_templatesByHandler, _serviceProvider);
+
+    private static readonly string[] _routes = [
```

Comment:
Can this be dynamically generated to test different numbers of routes and segments? It might also be a good idea to include some routes with `{routeValues}` even though the comparison ignores them.